scripts/autoclose.js

```diff
@@ -0,0 +1,159 @@
+import process from 'node:process';
+
+/**
+ * @typedef {object} Rule
+ * @property {RegExp[]} patterns - The pattern to match against the issue title.
+ * @property {string} reason - The issue numbers to include in the reason text.
+ */
+
+/**
+ * @typedef {object} Issue
+ * @property {{name: string}[]} labels - Isesue labels.
+ * @property {string} state - Issue state, possible values are 'open' and 'closed'.
+ * @property {string} title - Issue title.
+ * @property {string} body - Issue body.
+ */
+
+/** @type {Rule[]} */
+const rules = await import(
+	path.join(import.meta.dirname, '..', 'autoclose.rules.mjs')
+).then((module) => module.default);
+
+const {GITHUB_TOKEN, GITHUB_REPOSITORY, ISSUE_NUMBER} = process.env;
+
+if ([GITHUB_TOKEN, GITHUB_REPOSITORY, ISSUE_NUMBER].some((v) => !v)) {
+	console.error(
+		`${Object.entries({GITHUB_TOKEN, GITHUB_REPOSITORY, ISSUE_NUMBER})
+			.filter(([, v]) => !v)
+			.map(([k]) => k)
+			.join(', ')} environment variable(s) must be set.`,
+	);
+	process.exit(1);
+}
+
+/**
+ * Fetch data from GitHub API.
+ * @param {string} url The URL to fetch.
+ * @param {globalThis.RequestInit} options The options to pass to the fetch function.
+ * @returns {Promise<any>} - The response data.
+ */
+const githubFetch = async (url, options) => {
+	const response = await globalThis.fetch(new URL(url), {
+		...options,
+		headers: {
+			Accept: 'application/vnd.github+json',
+			Authorization: `Bearer ${GITHUB_TOKEN}`,
+			'X-GitHub-Api-Version': '2022-11-28',
+			...options?.headers,
+		},
+	});
+	if (!response.ok) {
+		throw new Error(
+			`Failed to fetch: ${response.status} (${response.statusText}).`,
+		);
+	}
+
+	return response;
+};
+
+/**
+ * Check if the issue is a won't add icon issue.
+ * GitHub REST API: https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#get-an-issue.
+ * @returns {Promise<string | undefined>} - Returns reason if the issue is a won't add icon issue, undefined otherwise.
+ */
+const checkIfCanBeClosed = async () => {
+	const url = `https://api.github.com/repos/${GITHUB_REPOSITORY}/issues/${ISSUE_NUMBER}`;
+	const response = await githubFetch(url, {method: 'GET'});
+	if (!response.ok) {
+		throw new Error(
+			`Failed to fetch: ${response.status} (${response.statusText}).`,
+		);
+	}
+
+	/** @type {Issue} */
+	const json = await response.json();
+	const {labels, state, title, body} = json;
+	const labelNames = new Set(labels.map((label) => label.name));
+	if (state === 'closed') return undefined;
+	if (labelNames.has('meta')) return undefined;
+	if (labelNames.has('in discussion')) return undefined;
+	if (!labelNames.has('new icon')) return undefined;
+	const matched = rules.find((rule) =>
+		rule.patterns.some((pattern) => {
+			const brandNamePattern = new RegExp(
+				`### Brand Name\n*${pattern.source.replaceAll('$', '')}\n*###`,
+				'i',
+			);
+			return pattern.test(title) || brandNamePattern.test(body);
+		}),
+	);
+	if (!matched) return undefined;
```

Comment:
```suggestion
	if (!matched) return;
```
Is there a reason (lint rule?) we are specifying an explicit undefined return value here and above?

Reply:
We should return `undefined` for more clear intention. And `return` only is more for void rather than `undefined`.